src/main/java/com/snowflake/kafka/connector/internal/streaming/SnowflakeSinkServiceV2.java

```diff
@@ -176,7 +177,33 @@ public void startTask(String tableName, TopicPartition topicPartition) {
     createTableIfNotExists(tableName);
 
     // Create channel for the given partition
-    createStreamingChannelForTopicPartition(tableName, topicPartition);
+    createStreamingChannelForTopicPartition(tableName, topicPartition, null);
+  }
+
+  /**
+   * Initializes multiple Channels and partitionsToChannel maps with new instances of {@link
+   * TopicPartitionChannel}
+   *
+   * @param partitions collection of topic partition
+   * @param topic2Table map of topic to table name
+   */
+  @Override
+  public void startTasks(Collection<TopicPartition> partitions, Map<String, String> topic2Table) {
+    Map<String, Boolean> tableName2SchemaEvolutionPermission = new HashMap<>();
```

Comment:
@sfc-gh-japatel @sfc-gh-rcheng Is there another place we could put this so that the map doesn't have to be repopulated for every rebalance? Could you move it to the place where we create the client? We should have all the table info from the configuration?

Reply:
The client is lazily created in the [SnowflakeSinkTask](https://github.com/snowflakedb/snowflake-kafka-connector/blob/master/src/main/java/com/snowflake/kafka/connector/internal/streaming/SnowflakeSinkServiceV2.java#L158) since we do need the configuration. It would not be affected by rebalance because the [singleton provider](https://github.com/snowflakedb/snowflake-kafka-connector/blob/master/src/main/java/com/snowflake/kafka/connector/internal/streaming/StreamingClientProvider.java)

Reply:
even if we move it to connector level, it is very much possible it needs to repopulated there. that's what is happening with client creation as well.

Reply:
Not sure if I understand, so do we have a better place to put this map so that it won't be repopulated every time after rebalance?

Reply:
I see your concern, valid point. 

I think this can be moved up a level inside a constructor of SnowflakeSinkServiceV2 and we can pass down the map. Is that what you are thinking?

Reply:
What you can do here is call `hasSchemaEvolutionPermission` from inside the Ctor of SnowflakeSinkServiceV2 for every table from topicToTable Map 

pass down the cache in startTask and do a get or if there is cache miss, populate the cache.